lib/client/doc.js

```diff
@@ -620,28 +621,28 @@ Doc.prototype._otApply = function(op, source) {
           if (transformErr) return this._hardRollback(transformErr);
         }
         // Apply the individual op component
-        this.emit('before op', componentOp.op, source, op.src);
+        this.emit('before op', componentOp.op, source, op.src, op.m);
```

Comment:
Would you be open to this signature, where the 3rd param is op.src, and 4th param is the whole op? This way whoever needs whatever from op have access to that whatever?
```javascript
this.emit('before op', componentOp.op, source, op.src, op);
```

Reply:
I think @ericyhwang had suggested something similar to this in the past. I'm not necessarily against it, although I'd potentially still bury it in a wrapper object in case we need more arguments